### ISSUE TITLE
Every filter not returning expected results

### ISSUE THREAD

**Issue by ElricLai** [prisma/prisma#25323]
### Bug description

Query
```
await prisma.state.findMany({
    where: {
        option: {
            **every: {
                storeId: [some-store-id]
            }**
        }
    }, {
        include: { state: true }
    }
});
```

State results
```
[
    {
        id: [some-id],
        store: []
    }, {
        id: [some-id],
        store: [
            { id: [some-id], storeId: [some-store-id] }
        ]
    }
]
```

Every filter should return where storeId matching instead of returning empty array of store

This should not be returned
```
{
    id: [some-id],
    store: []
}
```

### How to reproduce

prisma every filter with empty results

### Expected behavior

```
[
    {
        id: [some-id],
        store: [
            { id: [some-id], storeId: [some-store-id] }
        ]
    }
]
```

### Prisma information

```
model State {
    id   String @id @default(uuid())
    option Option[]
}

model StoreDeliveryOption {
  id             String       @id @default(uuid())
  destination    State     @relation(fields: [stated], references: [id], onDelete: Restrict)
  stateId     String       @map("state_id")
}
```

### Environment & setup

- Database: postgresql
- Remix version: v6

### Prisma Version

```
    "@prisma/client": "^5.17.0",
```

**Comment by SevInf** [comment]
Hi @ElricLai.
Thank you for opening the issue. To be honest, I have trouble following here. Could you please provide runnable reproduction as well as the exact data you expect it to return

**Comment by ElricLai** [comment]
@SevInf Thanks for your reply.

The schema i have is as below.. As we can see the relationship is:
  1. One state has many delivery options
  2. Each delivery option has one state

```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["strictUndefinedChecks"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model State {
    id   String @id @default(uuid())
    name String
    deliveryOptions DeliveryOption[]
}

model DeliveryOption {
  id             String       @id @default(uuid())
  name String
  storeId String
  destination    State     @relation(fields: [stateId], references: [id], onDelete: Restrict)
  stateId     String       @map("state_id")
}
```

When i tried to run the query:
```
await prisma.state.findMany({
    include: { deliveryOptions: true }
});
```

The query returns the results below:
```
[
    {
        id: "3ee9916f-5d98-487f-8297-06013f0854bf",
        name: "City A",
        deliveryOptions: [
            { id: "f71da1f6-f097-4a2e-9fb1-70a9e7f00368", name: "Standard", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
            { id: "f710d0f9-1fac-48f3-ae74-414896056785", name: "Express", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
        ]
    }, {
        id: "682e2bba-3200-45bd-ae14-b58ca417483d",
        name: "City B",
        deliveryOptions: [
            { id: "f71da1f6-f097-4a2e-9fb1-70a9e7f00368", name: "Standard", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
            { id: "f710d0f9-1fac-48f3-ae74-414896056785", name: "Express", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
        ]
    }, {
        id: "46081fb9-1a2a-44ca-a8f5-3377784d13fb",
        name: "City C",
        deliveryOptions: []
    }
]
```

Now, this is where the **bug** that I discovered. When I tried to run the query
```
await prisma.state.findMany({
    where: {
        deliveryOptions: {
            every: {
                storeId: "c9871b09-ce04-42ee-b27e-e01828356f66"
            }
        }
    },
    include: { deliveryOptions: true }
});
```
The same results returned:
```
[
    {
        id: "3ee9916f-5d98-487f-8297-06013f0854bf",
        name: "City A",
        deliveryOptions: [
            { id: "f71da1f6-f097-4a2e-9fb1-70a9e7f00368", name: "Standard", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
            { id: "f710d0f9-1fac-48f3-ae74-414896056785", name: "Express", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
        ]
    }, {
        id: "682e2bba-3200-45bd-ae14-b58ca417483d",
        name: "City B",
        deliveryOptions: [
            { id: "f71da1f6-f097-4a2e-9fb1-70a9e7f00368", name: "Standard", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
            { id: "f710d0f9-1fac-48f3-ae74-414896056785", name: "Express", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
        ]
    }, {
        id: "46081fb9-1a2a-44ca-a8f5-3377784d13fb",
        name: "City C",
        deliveryOptions: []
    }
]
```

But which the result should have excluded City C where storeId do not match anything
Expected results:
```
[
    {
        id: "3ee9916f-5d98-487f-8297-06013f0854bf",
        name: "City A",
        deliveryOptions: [
            { id: "f71da1f6-f097-4a2e-9fb1-70a9e7f00368", name: "Standard", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
            { id: "f710d0f9-1fac-48f3-ae74-414896056785", name: "Express", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
        ]
    }, {
        id: "682e2bba-3200-45bd-ae14-b58ca417483d",
        name: "City B",
        deliveryOptions: [
            { id: "f71da1f6-f097-4a2e-9fb1-70a9e7f00368", name: "Standard", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
            { id: "f710d0f9-1fac-48f3-ae74-414896056785", name: "Express", storeId: "c9871b09-ce04-42ee-b27e-e01828356f66" },
        ]
    }
]
```

**Comment by nurul3101** [comment]
Related Issue: #11819

TLDR: This is an expected behaviour. You can check the linked issue to get additional details.

To get the behaviour you are expecting you can do something like this:
```ts
await prisma.state.findMany({
  where: {
    AND: [
      {
        deliveryOptions: {
          some: {} // Ensures at least one delivery option exists
        }
      },
      {
        deliveryOptions: {
          every: {
            storeId: "c9871b09-ce04-42ee-b27e-e01828356f66"
          }
        }
      }
    ]
  },
  include: { deliveryOptions: true }
});
```
